Improve subspaces and some utilities around room/space creation

t3chguy opened the pull request:
Fixes https://github.com/vector-im/element-web/issues/18090
Fixes https://github.com/vector-im/element-web/issues/18091
Fixes https://github.com/vector-im/element-web/issues/17256

![image](https://user-images.githubusercontent.com/2403652/126748787-8a306229-3a3f-46c8-868a-3cd94080a345.png)
![image](https://user-images.githubusercontent.com/2403652/126748836-6126fcad-062f-4cfe-bee6-d210ae7bc9a8.png)
![image](https://user-images.githubusercontent.com/2403652/126751069-327632cf-acce-486d-b19b-5ab5be62f43a.png)

gsouquet commented on src/components/views/dialogs/CreateSubspaceDialog.tsx:
```diff
@@ -0,0 +1,183 @@
+            </div>
+
+            <div className="mx_CreateSubspaceDialog_footer">
+                <span>
```
I've noticed we're using `span` in a lot of places and it has block level children which is invalid HTML.
That create some UI weirdness for me in the past, we should try to avoid that where possible

t3chguy replied:
Good point, here it shouldn't matter but will update for sanity

gsouquet reviewed (approved):
Overall looks great to me 👏

niquewoodhouse commented:
Review:

**Medium things**

- The add existing spaces dialog seems to have a rogue No results item

<img src="https://user-images.githubusercontent.com/1587166/127192861-92ec46a6-e395-4383-a4c8-4edc4fbe73b1.png" width="300">

- When I added an existing space from the 'add' button on the explore space page, the added space disappears from the spaces panel  (makes sense) and the rooms within it are absorbed into the room list (makes sense), the subspace got added to the spaces panel (makes sense), but the 'explore space view' didn't update (only the room list did).  


- Missing microcopy on the add space dialog (I think this was my oversight)

<img src="https://user-images.githubusercontent.com/1587166/127199544-d57ebfc1-46f7-4fc4-a32e-9fe281786f2c.png" width="400">

We use this elsewhere, I think it's actually as useful here as anywhere else.

If its set to private (invite only), copy says "Only inviting people will be able to find and join"; if public "anyone in Space name will be able to find and join."
 

**Small things**

<img src="https://user-images.githubusercontent.com/1587166/127191605-90bc9717-881b-4b4e-998a-9fd325caff6b.png" width="400">

Would be nice if the text between the badge and the supporting copy could align better.   eg 

<img src="https://user-images.githubusercontent.com/1587166/127192216-e1bda9b0-e3fb-4ec0-8145-206d04bc6dbf.png" width="400">

t3chguy commented:
> When I added an existing space from the 'add' button on the explore space page, the added space disappears from the spaces panel (makes sense) and the rooms within it are absorbed into the room list (makes sense), the subspace got added to the spaces panel (makes sense), but the 'explore space view' didn't update (only the room list did).

Fixed

> Missing microcopy on the add space dialog (I think this was my oversight)

Added

![image](https://user-images.githubusercontent.com/2403652/127375671-263b300d-348d-4c62-9727-e90152453ba7.png)

> Would be nice if the text between the badge and the supporting copy could align better. eg

Fixed

![image](https://user-images.githubusercontent.com/2403652/127374301-1e2de593-e723-4c2a-a436-4fcaa86c0dce.png)

niquewoodhouse reviewed (changes requested):
![image](https://user-images.githubusercontent.com/1587166/127377073-5ce1d11a-5944-47fc-9343-b1faf14b6a88.png)

Copy is...not true, or is it?  I'm not sure but...are these subspaces just going to be set to just members of the space, or anyone anywhere?  Rooms will have 3 options - space members, private and public - these just have the final 2?

nadonomy commented:
Are we confident we want to expose subspace as nomenclature? It’s ringing
off jargon alarm bells for me. Wondering if the UI should simply stick to
spaces for now?

On Wed, 28 Jul 2021 at 19:35, Nique Woodhouse ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> [image: image]
> <https://user-images.githubusercontent.com/1587166/127377073-5ce1d11a-5944-47fc-9343-b1faf14b6a88.png>
>
> Copy is...not true, or is it? I'm not sure but...are these subspaces just
> going to be set to just members of the space, or anyone anywhere? Rooms
> will have 3 options - space members, private and public - these just have
> the final 2?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/matrix-org/matrix-react-sdk/pull/6458#pullrequestreview-717345276>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDJTMJWDQ6JZZZVPP7IOJLT2BEYLANCNFSM5A3P7BOQ>
> .
>

niquewoodhouse commented:
> Are we confident we want to expose subspace as nomenclature? It’s ringing off jargon alarm bells for me. Wondering if the UI should simply stick to spaces for now?

I don't feel super strongly.  The original designs had space too but in the delight team, people got confused as to what it meant, so changed it.  My theory was a folder in a folder is a folder, not a subfolder but I have seen some external people in the feedback room refer to them as 'subspaces' so it makes me think it's possibly understood, but that might be just be more technical people who picked up the term from the room?

t3chguy commented:
> Copy is...not true, or is it? I'm not sure but...are these subspaces just going to be set to just members of the space, or anyone anywhere? Rooms will have 3 options - space members, private and public - these just have the final 2?


> - these just have the final 2?

Why? 

What if you have an organisation e.g Element but want a public subspace for public relations
or what if you're a public space like Matrix Community, are you going to force everyone to join Matrix Community to be able to join any subspaces?

Given that you can create a public space and add it as a subspace to another space this is a possible scenario by jumping through UX hoops anyway.

> subspaces

Happy to change all references to Spaces if that is better

niquewoodhouse commented:
> > * these just have the final 2?
> 
> Why?

Just to clarify, I was asking why they only have 2 options and not the 3 options as intended.  And then you reminded me it's because my server doesn't yet have the restricted rooms upgrade, which makes sense, thanks.

niquewoodhouse commented:
@nadonomy @t3chguy can we go ahead with this plan: Change subspace to space and see what reaction we get from people, ahead of exiting beta, and just make a decision at that point what we do?

t3chguy commented:
Updated copy
